### PR TITLE
add Box3

### PR DESCRIPTION
**Pull request by frederikja163**
### Purpose of this PR

* Add box3
* Affects OTK.Math

### Testing status

* No tests planned yet

This PR just adds the box3, and is gonna get cleaned up further along with box2 in #920

### CODE REVIEW

File: src/OpenToolkit.Mathematics/Geometry/Box3.cs [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+
+        /// <summary>
+        /// The back boundary of the structure
+        /// </summary>
+        public float Back;
+
+        /// <summary>
+        /// The front boundary of the structure
+        /// </summary>
+        public float Front;
```
Comment (jvbsl) [inline src/OpenToolkit.Mathematics/Geometry/Box3.cs]: with these we only have AABB, so would min/max Vector3 not a lot easier to use and access?

File: src/OpenToolkit.Mathematics/Geometry/Box3.cs [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+            Bottom = bottom;
+            Back = back;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Box3"/> struct.
+        /// </summary>
+        /// <param name="frontTopLeft">An OpenToolkit.Vector3 describing the front-top-left corner of the Box3.</param>
+        /// <param name="backBottomRight">An OpenToolkit.Vector3 describing the back-bottom-right corner of the Box3.</param>
+        public Box3(Vector3 frontTopLeft, Vector3 backBottomRight)
```
Comment (jvbsl) [inline src/OpenToolkit.Mathematics/Geometry/Box3.cs]: min/max are a lot easier to understand in my honest opinion

File: src/OpenToolkit.Mathematics/Geometry/Box3.cs (hunk 1 above)
Comment (frederikja163) [reply src/OpenToolkit.Mathematics/Geometry/Box3.cs]: Will be fixed in #920

File: src/OpenToolkit.Mathematics/Geometry/Box3.cs (hunk 2 above)
Comment (frederikja163) [reply src/OpenToolkit.Mathematics/Geometry/Box3.cs]: Will get fixed in #920

**Pull request merged**